onnx/optimizer/pass_manager.cc

```diff
@@ -0,0 +1,46 @@
+#include "onnx/optimizer/pass_manager.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+PassManager::PassManager() {}
+PassManager::~PassManager() {}
+
+GeneralPassManager::~GeneralPassManager() {
+  this->passes.clear();
+}
+void GeneralPassManager::add(std::shared_ptr<Pass> pass) {
+  this->passes.insert(pass);
+}
+
+std::shared_ptr<PassManagerAnalysis> GeneralPassManager::run(Graph& graph) {
+  for (std::shared_ptr<Pass> pass : this->passes) {
+    auto pass_analysis = pass->runPass(graph);
+  }
+  return std::shared_ptr<PassManagerAnalysis>(new EmptyPassManagerAnalysis());
+}
+
+std::shared_ptr<PassManagerAnalysis> FixedPointPassManager::run(Graph& graph) {
+  bool fixed_point_optimization_done;
+
+  do {
+    fixed_point_optimization_done = false;
+    for (std::shared_ptr<Pass> pass : this->passes) {
+      std::shared_ptr<PostPassAnalysis> analysis = pass->runPass(graph);
+      std::shared_ptr<CountBasedPassAnalysis> count_analysis =
+          std::dynamic_pointer_cast<CountBasedPassAnalysis>(analysis);
```

Comment:
Can we get ride of `dynamic_pointer_cast`?

Reply:
How should we circumvent the use of RTTI? We need to know the type of analysis done in order to know whether or not to do fixed point optimization and this cannot be inferred at a compile time right?

Reply:
Can you add one field in the parent class? For different type of children, you assign with different values, and do the static cast at runtime. Does that sound good to you?

Reply:
Good idea. I'm working on the revisions now.